src/components/liveSession/components/liveSessionBtn/liveSessionBtn.component.js

```diff
@@ -98,29 +101,82 @@
 
                 this.getScheduledLesson = () => {
                     let dataPromMap = {
-                        liveSessionDuration: this.liveSessionSettings ? $q.when(this.liveSessionSettings) : liveSessionSettingsProm,
+                        liveSessionSettings: this.liveSessionSettings ? $q.when(this.liveSessionSettings) : liveSessionSettingsProm,
                         educatorProfile: this.educatorProfile ? $q.when(this.educatorProfile) : educatorProfileProm
                     };
                     return $q.all(dataPromMap).then(dataMap => {
                         this.liveSessionSettings = dataMap.liveSessionSettings;
                         this.educatorProfile = dataMap.educatorProfile;
                         SESSION_DURATION = this.liveSessionSettings ? this.liveSessionSettings : SESSION_DURATION;
                         let now = Date.now();
-                        let calcStartTime = now - SESSION_DURATION.marginBeforeSessionStart;
-                        let calcEndTime = now + SESSION_DURATION.marginAfterSessionStart;
+                        let calcStartTime = now - SESSION_DURATION.queryLessonStart;
+                        let calcEndTime = now + SESSION_DURATION.length + SESSION_DURATION.queryLessonEnd;
                         let dateRange = {
                             startDate: calcStartTime,
                             endDate: calcEndTime
                         };
 
                         return ZnkLessonNotesSrv.getLessonsByStudentIds([this.student.uid], dateRange, this.educatorProfile.uid)
                             .then(lessons => {
-                                return lessons.data && lessons.data.length ? lessons.data[0] : null;
-                            }, err => $log.debug('getScheduledLesson: getLessonsByStudentIds Error: ', err));
+                                return this.getLessonInRange(lessons.data);
+                            }, err => $log.error('getScheduledLesson: getLessonsByStudentIds Error: ', err));
                     });
 
 
                 };
+
+                this.getLessonInRange = (lessons) => {
+                    let scheduledLesson = null;
+                    if (lessons && lessons.length > 0) {
+                        if (lessons.length === 1) {
+                            scheduledLesson = lessons.pop();
+                        } else {
+                            $log.debug(`getLessonInRange: multiple lesson - check if it's back to back`);
+                            scheduledLesson = this.checkBack2BackLesson(lessons);
+                        }
+                    }
+                    $log.debug(`getLessonInRange: lessons is not defined or lessons.length === 0 `);
+                    return scheduledLesson;
```

Comment:
instead of returning scheduledLesson, 
return:
{ 
  scheduledLesson: scheduledLesson,
  expectedSessionEndTime: number
}